x/programs/cmd/simulator/src/lib.rs

```diff
@@ -223,109 +225,99 @@ impl Client {
         Self { path }
     }
 
+    fn read_response(
+        lines: &mut Lines<BufReader<&mut ChildStdout>>,
+    ) -> Result<PlanResponse, Box<dyn Error>> {
+        let text: String = lines
+            .next()
+            .ok_or("EOF")?
+            .map_err(|e| format!("failed to read from stdout: {e}"))?;
+
+        let resp = serde_json::from_str(&text)
+            .map_err(|e| format!("failed to parse output to json: {e}"))?;
+
+        Ok(resp)
+    }
+
+    fn write_stdin(stdin: &mut ChildStdin, bytes: &[u8]) -> Result<(), String> {
```

Comment:
Same with this, there's no reason to have this as a static method.

Reply:
Moving this out